nixos/modules/programs/starship.nix

```diff
@@ -0,0 +1,62 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.programs.starship;
+
+in {
+  meta.maintainers = [ maintainers.imalison ];
+
+  options.programs.starship = {
+    enable = mkEnableOption "starship";
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.starship;
+      defaultText = literalExample "pkgs.starship";
+      description = "The package to use for the starship binary.";
+    };
+
+    enableBashIntegration = mkOption {
+      default = true;
+      type = types.bool;
+      description = ''
+        Whether to enable Bash integration.
+      '';
+    };
+
+    enableZshIntegration = mkOption {
+      default = true;
+      type = types.bool;
+      description = ''
+        Whether to enable Zsh integration.
+      '';
+    };
+
+    enableFishIntegration = mkOption {
```

Comment:
Forgot to add `fish.promptInit`?

Reply:
No, I meant to remove this.

Reply:
May I ask why? :)

Reply:
Was annoying to write the emacs condition in fish, and i got lazy.

Reply:
I just don't know fish well, and didnt really want to take the time to learn it. You are welcome to write something equivalent and I will add it back.